Svc/FileManager/FileManager.cpp

```diff
@@ -156,10 +164,106 @@ void FileManager ::FileSize_cmdHandler(const FwOpcodeType opCode, const U32 cmdS
+    m_totalEntries = 0;
+    
+    // Directory listing will be processed asynchronously by Rate Group 2 (0.5Hz).
+    // The schedIn_handler will process one directory entry per rate tick to
+    // prevent event flooding and ensure bounded execution time.
+    // Command response will be sent when listing completes.
+}
+
 void FileManager ::pingIn_handler(const FwIndexType portNum, U32 key) {
     // return key
     this->pingOut_out(0, key);
 }
+
+void FileManager ::schedIn_handler(const FwIndexType portNum, U32 context) {
+    // Only process if we're in the middle of a directory listing
+    if (m_listState == LISTING_IN_PROGRESS) {
+        // Process one file per rate tick
+        Fw::String filename;
+        Os::Directory::Status status = m_currentDir.read(filename);
+        
+        if (status == Os::Directory::NO_MORE_FILES) {
+            // We're done listing - close directory and send response
+            m_currentDir.close();
+            m_listState = IDLE;
+            
+            Fw::LogStringArg logStringDirName(m_currentDirName.toChar());
+            this->log_ACTIVITY_HI_ListDirectorySucceeded(logStringDirName, m_totalEntries);
+            this->emitTelemetry(Os::FileSystem::OP_OK);
+            this->sendCommandResponse(m_currentOpCode, m_currentCmdSeq, Os::FileSystem::OP_OK);
+            
+        } else if (status == Os::Directory::OP_OK) {
+            // Successfully read a file - emit event for this file
+            Fw::LogStringArg logStringDirName(m_currentDirName.toChar());
+            Fw::LogStringArg logStringFileName(filename.toChar());
+            
+            // Construct full path for type checking
+            Fw::String fullPath;
+            fullPath.format("%s/%s", m_currentDirName.toChar(), filename.toChar());
+            
+            // Determine entry type
+            Os::FileSystem::PathType pathType = Os::FileSystem::getPathType(fullPath.toChar());
+            
+            if (pathType == Os::FileSystem::FILE) {
+                // Regular file: get size and emit file event
+                FwSizeType fileSize;
+                Os::FileSystem::Status sizeStatus = Os::FileSystem::getFileSize(fullPath.toChar(), fileSize);
+                U64 displaySize = (sizeStatus == Os::FileSystem::OP_OK) ? static_cast<U64>(fileSize) : 0;
+                this->log_ACTIVITY_HI_DirectoryListing(logStringDirName, logStringFileName, m_currentIndex, displaySize);
+            } else if (pathType == Os::FileSystem::DIRECTORY) {
+                // Subdirectory: emit subdirectory event with file count 0 (simplified)
+                this->log_ACTIVITY_HI_DirectoryListingSubdir(logStringDirName, logStringFileName, m_currentIndex, 0);
+            } else {
+                // Special file or inaccessible: treat as file with 0 size
+                this->log_ACTIVITY_HI_DirectoryListing(logStringDirName, logStringFileName, m_currentIndex, 0);
+            }
+            
+            m_currentIndex++;
```

Comment:
Do we need the index of the file listing?  Seems like it is extra unneeded information.